tests/plugins/softlayer_test.py

```diff
@@ -0,0 +1,167 @@
+            ('sl-apikey: "insert_key_here"', False),
+            ('softlayer-key:=afakekey', False),
+            ('fake-softlayer-key= "not_long_enough"', False),
+        ],
+    )
+    def test_analyze_string(self, payload, should_flag):
+        logic = SoftlayerDetector()
+
+        output = logic.analyze_string(payload, 1, 'mock_filename')
+        assert len(output) == (1 if should_flag else 0)
+
+    @responses.activate
+    def test_verify_invalid_secret(self):
+        responses.add(
+            responses.GET, 'https://api.softlayer.com/rest/v3/SoftLayer_Account.json',
+            json={'error': 'Access denied. '}, status=401,
+        )
+
+        assert SoftlayerDetector().verify(
+            SL_TOKEN,
+            'softlayer_username={}'.format(SL_USERNAME),
+        ) == VerifiedResult.VERIFIED_FALSE
+
+    @responses.activate
+    def test_verify_valid_secret(self):
+        responses.add(
+            responses.GET, 'https://api.softlayer.com/rest/v3/SoftLayer_Account.json',
+            json={'id': 1}, status=200,
+        )
+        assert SoftlayerDetector().verify(
+            SL_TOKEN,
+            'softlayer_username={}'.format(SL_USERNAME),
+        ) == VerifiedResult.VERIFIED_TRUE
+
+    @responses.activate
+    def test_verify_unverified_secret(self):
+        assert SoftlayerDetector().verify(
+            SL_TOKEN,
+            'softlayer_username={}'.format(SL_USERNAME),
+        ) == VerifiedResult.UNVERIFIED
+
+    def test_verify_no_secret(self):
+        assert SoftlayerDetector().verify(
+            SL_TOKEN,
+            'no_un={}'.format(SL_USERNAME),
+        ) == VerifiedResult.UNVERIFIED
+
+    @pytest.mark.parametrize(
+        'content, expected_output',
+        (
+            (
+                textwrap.dedent("""
+                    --softlayer-username = {}
+                """)[1:-1].format(
+                    SL_USERNAME,
+                ),
+                [SL_USERNAME],
+            ),
+
+            # With quotes
+            (
+                textwrap.dedent("""
+                    sl_user_id = "{}"
+                """)[1:-1].format(
+                    SL_USERNAME,
+                ),
+                [SL_USERNAME],
+            ),
+
+            # multiple candidates
+            (
+                textwrap.dedent("""
+                    softlayer_id = '{}'
+                    sl-user = '{}'
+                    SOFTLAYER_USERID = '{}'
+                    softlayer-uname: {}
+                """)[1:-1].format(
+                    SL_USERNAME,
+                    'test2@testy.test',
+                    'test3@testy.testy',
+                    'notanemail',
+                ),
+                [
+                    SL_USERNAME,
+                    'test2@testy.test',
+                    'test3@testy.testy',
+                    'notanemail',
+                ],
+            ),
+        ),
+    )
+    def test_find_username(self, content, expected_output):
```

Comment:
Just in case you find that this type of secret is used in function calls, the following commit may help you add detection for that. (Code still in review --  https://github.com/Yelp/detect-secrets/pull/255/commits/c6037d998fc3b03fb2662b1df178a062d6566705)

Reply:
Unlike AWS tokens which has a specific format, the Softlayer username here is almost free format. It might be hard to match them in function calls. I will prefer to leave them out for now.

Reply:
Thanks for the suggestions. @KevinHock. Unlike AWS tokens which has a specific format, the Softlayer username here is almost free format. It might be hard to match them in function calls. I will prefer to leave them out for now.